### DISCUSSION TITLE
Can I use the withCredential option to load GLTF files ?

### DISCUSSION THREAD

**Discussion by ymoriaud** [google/model-viewer#3236] [Q&A] [answered]
Hi everyone 👋 

I have a question regarding about file of the 3D model, that are accessible with an authentication cookie.

I know that three.js loaders has the `withCredential` option on it (https://threejs.org/docs/#api/en/loaders/Loader.withCredentials), allowing the request to be sent with cookies.

I was wondering if there is a way from model-viewer to setup this option ?

Thanks for your answers, I you have any questions, let me know !

**Comment by samaneh-kazemi** [comment]
Hi @ymoriaud, thanks for reaching out! We currently don't support such a feature. Can you please tell us more about your use case?

**Reply by ymoriaud** [reply]
Thanks @samaneh-kazemi for your quick answer !

> We currently don't support such a feature. 

Understood. :)

> Can you please tell us more about your use case?

Yes of course ! I'm using the model-viewer tool to access non public .GLTF files. Files are served by a static server that can only be accessible with a specific HTTP cookie to authenticate client requests.
The browser is taking care of the cookie if the `withCredential` attribute is provided by the client.
Does this make sense to you ? :)

I can try to draft a PR. I tried to locate where the threejs loader is used, by I'm not sure I found it. If you can point me to a direction, I can try somethings.

**Reply by samaneh-kazemi** [reply]
@ymoriaud Thanks for sharing! :) Start with looking at `CachingGLTFLoader.ts`. And a PR is always encouraged! Thanks for contributing!

**Reply by ymoriaud** [reply]
Thanks @samaneh-kazemi ! I will look at this file. :)

**Comment by ymoriaud** [comment, marked as answer]
Hi ! :wave:

I created an new PR linked to this discussion ➡️  **Add with-credentials attribute to configure three.js loader** #3239

**Reply by ymoriaud** [reply]
PR has been merges. Thanks everyone. :)

**Reply by felipenmoura** [reply]
Hi.
Awesome to see this is supported.
I am facing exactly the same situation, where I have to load non-public .blg assets.

BUT ... I am not sure the root cause is this...
That's because for some reason, I can't see the request in the browser's network tab!
If I use the url of a public asset, it works just fine...when I use the url of a non-public file it does not, but it doesn't even trigger the fetch call (or so it seems). No `onError` or `onLoad` is triggered either...it just sits there, empty.

Any idea what could be causing it?
Thanks.

**Reply by elalish** [reply]
What's a `.blg`? We only support `.gltf` and `.glb` 3D models, so that's the most logical reason why we don't try to fetch anything.

**Reply by felipenmoura** [reply]
Well, I think I figured out the reason.

I was using a signedUrl to load the file and it looks like there is something related to cross domain, then.
I am now using a URL within my domain that redirects to the signedURL (after validating the tokens and all) and it seems to work...I will test some other cases.
